hubconf.py

```diff
@@ -1,69 +1,57 @@
-            nar_ckpt = torch.hub.load_state_dict_from_url(
-                nar_url, progress=progress, check_hash=False, map_location='cpu'
-            )
-    else: nar_ckpt = torch.load(str(nar_path), map_location='cpu')
-    logging.info("Initializing modules...")
-    mars5 = Mars5TTS(ar_ckpt, nar_ckpt, device=device)
-    return mars5, InferenceConfig
+dependencies = ['torch', 'torchaudio', 'numpy', 'vocos', 'safetensors']
 
+# Centralized checkpoint URLs for easy management and updates
+CHECKPOINT_URLS = {
+    "ar": "https://github.com/Camb-ai/MARS5-TTS/releases/download/v0.3/mars5_en_checkpoints_ar-2000000.pt",
+    "nar": "https://github.com/Camb-ai/MARS5-TTS/releases/download/v0.3/mars5_en_checkpoints_nar-1980000.pt",
+    "ar_sf": "https://github.com/Camb-ai/MARS5-TTS/releases/download/v0.3/mars5_en_checkpoints_ar-2000000.safetensors",
+    "nar_sf": "https://github.com/Camb-ai/MARS5-TTS/releases/download/v0.3/mars5_en_checkpoints_nar-1980000.safetensors"
+}
 
-def _load_safetensors_ckpt(url, progress):
-    """ Loads checkpoint from a safetensors file """
+def load_checkpoint(url, progress=True, ckpt_format='pt'):
+    """ Helper function to download and load a checkpoint, reducing duplication """
     hub_dir = torch.hub.get_dir()
     model_dir = os.path.join(hub_dir, 'checkpoints')
     os.makedirs(model_dir, exist_ok=True)
     parts = torch.hub.urlparse(url)
     filename = os.path.basename(parts.path)
     cached_file = os.path.join(model_dir, filename)
+
     if not os.path.exists(cached_file):
-        # download it
         torch.hub.download_url_to_file(url, cached_file, None, progress=progress)
-    # load checkpoint
+
+    if ckpt_format == 'safetensors':
+        return _load_safetensors_ckpt(cached_file)
+    else:
+        return torch.load(cached_file, map_location='cpu')
+
+def _load_safetensors_ckpt(file_path):
+    """ Loads a safetensors checkpoint file """
     ckpt = {}
-    with safe_open(cached_file, framework='pt', device='cpu') as f:
+    with safe_open(file_path, framework='pt', device='cpu') as f:
         metadata = f.metadata()
         ckpt['vocab'] = {'texttok.model': metadata['texttok.model'], 'speechtok.model': metadata['speechtok.model']}
-        ckpt['model'] = {}
-        for k in f.keys(): ckpt['model'][k] = f.get_tensor(k)
+        ckpt['model'] = {k: f.get_tensor(k) for k in f.keys()}
     return ckpt
+
+def mars5_english(pretrained=True, progress=True, device=None, ckpt_format='safetensors', ar_path=None, nar_path=None):
+    """ Load Mars5 English model on `device`, optionally show `progress`. """
+    if device is None:
+        device = 'cuda' if torch.cuda.is_available() else 'cpu'
+    logging.info(f"Using device: {device}")
+
+    if not pretrained:
+        raise ValueError('Only pretrained models are currently supported.')
+
+    ar_ckpt = load_checkpoint(CHECKPOINT_URLS[f'ar_{ckpt_format}'], progress, ckpt_format) if ar_path is None else torch.load(ar_path, map_location='cpu')
```

Comment:
in case a user provides a custom safetensors format checkpoint, could you perhaps modify this to also detect whether user-provided `ar_path` is a `.safetensors` file or a `.pt` file? Perhaps osmething like
```python
if ar_path is None:
     # same logic as before
elif ar_path is not None and ar_path.endswith('.pt'):
    ar_ckpt = torch.load(....)
elif ar_path is not None and ar_path.endswith('.safetensors'):
   ar_ckpt = # load with safetensors
else: raise NotImplementedError
```